### ISSUE TITLE
Two tile sets with same attributions shows up twice in GL JS.  iOS handles this case

### ISSUE THREAD

**Issue by roblabs** [mapbox/mapbox-gl-js#5788]
Use case — raster layer and vector layer are from two separate `.mbtiles` files.  In this screen grab, you can see the USGS attribution shows up twice.

<img width="479" alt="test-double-attributions" src="https://user-images.githubusercontent.com/118112/33498435-2eb82b7a-d686-11e7-9f04-18a2265e6837.png">

---

**mapbox-gl-js version**:
* 42.0
* Live demo [here](https://api.mapbox.com/styles/v1/roblabs/cjao9bejhfaua2spl9tpmvgo3.html?fresh=true&title=true&access_token=pk.eyJ1Ijoicm9ibGFicyIsImEiOiJwVlg0cnZnIn0.yhekddtKwZohGoORaWjqIw#11.33/42.1431/-79.2255) 

### Steps to Trigger Behavior

 1. Make two tile sets in `tippecanoe` or however you bake tiles
 2. Set attributions to be exactly the same
 3. Upload tile sets to Mapbox.com/studio
 4.  Make a new style and include these tile sets

### Actual Behavior
* two attributions are showing up

--- 

### Expected Behavior
* Using iOS as a precedent, only one attribution should show up

<img width="210" alt="ios" src="https://user-images.githubusercontent.com/118112/33498578-baf7df22-d686-11e7-9769-08593f8100af.png">

Originally posted to https://github.com/klokantech/tileserver-gl/issues/240

**Comment by jfirebaugh** [comment]
GL JS has code to deduplicate as well:

https://github.com/mapbox/mapbox-gl-js/blob/7151b2bef890ca585591edf081e8665a440c5fcc/src/ui/control/attribution_control.js#L125-L133

Can you help us figure out why this isn't working for your attribution strings?

**Comment by roblabs** [comment]
@jfirebaugh, after looking at the Javascript you posted, I looked into the logic and my attribution lengths were different.

My bad... my copyright character was defined two different ways.  Thanks for the pointer on GL JS handles this in the first place; closing as it was due to my attributions.

One of them with &copy;

``` javascript
"attribution": "<a href=\"http://www.usgs.gov\" target=\"_blank\">&copy; USGS</a>",
```

---
The other with unicode (fist shake)

``` javascript
"attribution": "<a href=\"https://usgs.gov\" target=\"_blank\">\u00a9 USGS</a>", 
```
